### PR TITLE
fix(input): pass through date input types, default config

### PR DESCRIPTION
**Pull request by MaximBalaganskiy**
Returning true is needed to support clicks on a native calendar button.

### CODE REVIEW

**Comment by bigopon** [comment]
I guess there's no reason to not pass this through. Though do we need to handle the value of date-* input differently? always get as Date object or just a string?

**Comment by MaximBalaganskiy** [comment]
Converting to a date would be convenient but not consistent with the native input. Not sure, personally

**Comment by bigopon** [comment]
> Converting to a date would be convenient but not consistent with the native input. Not sure, personally

we have `input.valueAsDate` for this, though I've never tried all `input`/`change` events combination with different browsers to be sure enough of this either.

File: packages/input/src/ux-input.ts [hunk 1]
```diff
@@ -8,12 +8,15 @@ import { UxInputTheme } from './ux-input-theme';
...
 // tslint:disable-next-line: no-submodule-imports
 import '@aurelia-ux/core/components/ux-input-component--outline.css';
+import { UxDefaultInputConfiguration } from './ux-default-input-configuration';
 
 export interface UxInputElement extends HTMLElement {
   value: any;
 }
 
-@inject(Element, StyleEngine)
+export type InputVariant = 'filled' | 'outline';
```
Comment (bigopon) [inline packages/input/src/ux-input.ts:17]: maybe this type should be exported from core, as we have `InputComponent` from there

File: packages/input/src/ux-input.ts [hunk 2]
```diff
@@ -43,8 +46,17 @@ export class UxInput implements UxInputComponent {
   public value: any;
   public textbox: HTMLInputElement;
 
-  constructor(private element: UxInputElement, public styleEngine: StyleEngine) {
+  constructor(private element: UxInputElement, public styleEngine: StyleEngine, private defaultConfiguration: UxDefaultInputConfiguration) {
```
Comment (bigopon) [inline packages/input/src/ux-input.ts:49]: if the configuration is not used anywhere else after the constructor, let's not assign it to the ux input

File: packages/input/src/ux-input.ts [hunk 3]
```diff
@@ -43,8 +46,17 @@ export class UxInput implements UxInputComponent {
...
     defineUxInputElementApis(element);
+    if (this.defaultConfiguration.theme !== undefined) {
```
Comment (bigopon) [inline packages/input/src/ux-input.ts:51]: @ben-girardet we now have default configuration for the ux input as well

File: packages/input/src/ux-input.ts (hunk 2 above)
Comment (MaximBalaganskiy) [reply packages/input/src/ux-input.ts]: done

File: packages/input/src/ux-input.ts (hunk 1 above)
Comment (MaximBalaganskiy) [reply packages/input/src/ux-input.ts]: done

**Pull request merged**